packages/ui/components/ui/tx/view/value/index.tsx

```diff
@@ -5,6 +5,19 @@ import { Amount } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/nu
 import { formatAmount } from '@penumbra-zone/types/amount';
 import { ValueComponent } from './value';
 
+// Uses exponent in metadata to display amount in terms of display denom
+export const getFormattedAmtFromValueView = (v: ValueView, withCommas = false): string => {
+  if (v.valueView.case === 'knownAssetId' && v.valueView.value.metadata) {
+    const { amount = new Amount(), metadata } = v.valueView.value;
+    const exponent = getDisplayDenomExponent.optional()(metadata);
+    return formatAmount({ amount, exponent, withCommas });
+  } else if (v.valueView.case === 'unknownAssetId') {
+    const { amount = new Amount() } = v.valueView.value;
+    return formatAmount({ amount, withCommas });
+  }
+  throw new Error(`Cannot derive formatted amount from value view: ${JSON.stringify(v.toJson())}`);
+};
```

Comment:
Extracted from below so it can be used outside

Reply:
Hm, can we put this somewhere else, like in `packages/types`? IMO, other components shouldn't be importing universal helpers from a single component's file.